pyvista/core/dataset.py

```diff
@@ -586,6 +586,183 @@ def translate(self, xyz: Union[list, tuple, np.ndarray]):
+        >>> from pyvista import examples
+        >>> pl = pyvista.Plotter(shape=(1, 2))
+        >>> pl.subplot(0, 0)
+        >>> pl.show_axes()
+        >>> _ = pl.show_grid()
+        >>> mesh1 = examples.download_teapot()
+        >>> _ = pl.add_mesh(mesh1)
+        >>> pl.subplot(0, 1)
+        >>> pl.show_axes()
+        >>> _ = pl.show_grid()
+        >>> mesh2 = mesh1.copy()
+        >>> mesh2.scale([10.0, 10.0, 10.0])
+        >>> _ = pl.add_mesh(mesh2)
+        >>> pl.show(cpos="xy")
+        """
+        self.points *= np.asarray(xyz)
+
+    def flip_x(self, point=None, transform_all_input_vectors=False):
+        """Flip mesh about the x-axis.
+
+        Parameters
+        ----------
+        point : list, optional
+            Point to rotate about.  Defaults to origin ``(0.0, 0.0, 0.0)``.
+
+        transform_all_input_vectors : bool, optional
+            When ``True``, all input vectors are
+            transformed. Otherwise, only the points, normals and
+            active vectors are transformed.
+
+        Examples
+        --------
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> pl = pyvista.Plotter(shape=(1, 2))
+        >>> pl.subplot(0, 0)
+        >>> pl.show_axes()
+        >>> mesh1 = examples.download_teapot()
+        >>> _ = pl.add_mesh(mesh1)
+        >>> pl.subplot(0, 1)
+        >>> pl.show_axes()
+        >>> mesh2 = mesh1.copy()
+        >>> mesh2.flip_x()
+        >>> _ = pl.add_mesh(mesh2)
+        >>> pl.show(cpos="xy")
+        """
+        if point is None:
+            point = (0.0, 0.0, 0.0)
+        if len(point) != 3:
+            raise ValueError('Point must be a vector of 3 values.')
+        t = transformations.reflection((1, 0, 0), point=point)
+        self.transform(t, transform_all_input_vectors=transform_all_input_vectors, inplace=True)
+
+    def flip_y(self, point=None, transform_all_input_vectors=False):
+        """Flip mesh about the y-axis.
+
+        Parameters
+        ----------
+        point : list, optional
+            Point to rotate about.  Defaults to origin ``(0.0, 0.0, 0.0)``.
+
+        transform_all_input_vectors : bool, optional
+            When ``True``, all input vectors are
+            transformed. Otherwise, only the points, normals and
+            active vectors are transformed.
+
+        Examples
+        --------
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> pl = pyvista.Plotter(shape=(1, 2))
+        >>> pl.subplot(0, 0)
+        >>> pl.show_axes()
+        >>> mesh1 = examples.download_teapot()
+        >>> _ = pl.add_mesh(mesh1)
+        >>> pl.subplot(0, 1)
+        >>> pl.show_axes()
+        >>> mesh2 = mesh1.copy()
+        >>> mesh2.flip_y()
+        >>> _ = pl.add_mesh(mesh2)
+        >>> pl.show(cpos="xy")
+        """
+        if point is None:
+            point = (0.0, 0.0, 0.0)
+        if len(point) != 3:
+            raise ValueError('Point must be a vector of 3 values.')
```

Comment:
Let's replace this with `check_valid_vector`.  This way we don't have to test it since it's already tested in a variety of other methods.

Reply:
```suggestion
        check_valid_vector(point, 'point')
```

Reply:
We also need to use `check_valid_vector` in rotate_* filters.